src/Workspaces/VisualBasic/Portable/Indentation/VisualBasicIndentationService.Indenter.vb

```diff
@@ -26,35 +26,40 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Indentation
 
         Protected Overrides Function GetDesiredIndentationWorker(
                 indenter As Indenter,
-                token As SyntaxToken,
-                previousLine As TextLine,
-                lastNonWhitespacePosition As Integer) As IndentationResult
+                tokenOpt As SyntaxToken?,
+                triviaOpt As SyntaxTrivia?) As IndentationResult
```

Comment:
vb has several special cases i've kept.  unfortunately, moving the code around has made this fairly messy.  please let me know if there are questions about this that i can answer.